### PR TITLE
Node notes cli

### PR DESCRIPTION
**Pull request by Quintin-Russell**
cli-assignment

### CODE REVIEW

File: node-notes-cli/main.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  dataJSON.notes = obj;
+  const data = JSON.stringify(dataJSON, null, 2);
+  return data
+}
+
+function update() {
+  dataJSON.notes[argument] = argument2;
+  const data = JSON.stringify(dataJSON, null, 2);
+  return data
+}
```
Comment (TimelikeClosure) [inline node-notes-cli/main.js:54]: I'd suggest making these more reusable by:
- Moving them into a separate module.
- Passing in any values they need.

File: node-notes-cli/main.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  }
+}
+
+function create() {
+  let nextID = dataJSON.nextId;
+  dataJSON.notes[nextID] = argument;
+  dataJSON.nextId = nextID + 1
+  const data = JSON.stringify(dataJSON, null, 2);
+  return data
+}
```
Comment (TimelikeClosure) [inline node-notes-cli/main.js:35]: What happens if the user forgets supply the contents for their new note?

File: node-notes-cli/main.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+  for (let key in dataJSON.notes) {
+    const val = dataJSON.notes[key];
+    if (key !== argument) {
+      obj[key] = val
+    }
+  }
+  dataJSON.notes = obj;
+  const data = JSON.stringify(dataJSON, null, 2);
+  return data
+}
```
Comment (TimelikeClosure) [inline node-notes-cli/main.js:48]: How would you let the user know if they try to delete a note that doesn't exist?

File: node-notes-cli/main.js (hunk 1 above)
Comment (TimelikeClosure) [inline node-notes-cli/main.js:54]: "User can update a note" shouldn't allow the user to create one by accident. How would you prevent that from happening?

**Review by TimelikeClosure — CHANGES_REQUESTED** [review]

File: node-notes-cli/main.js (hunk 1 above)
Comment (Quintin-Russell) [reply node-notes-cli/main.js:54]: edited

File: node-notes-cli/main.js (hunk 2 above)
Comment (Quintin-Russell) [reply node-notes-cli/main.js:35]: addressed

File: node-notes-cli/main.js (hunk 3 above)
Comment (Quintin-Russell) [reply node-notes-cli/main.js:48]: addressed

File: node-notes-cli/main.js (hunk 1 above)
Comment (Quintin-Russell) [reply node-notes-cli/main.js:54]: addressed

**Review by TimelikeClosure — APPROVED** [review]

**Pull request merged**